src/interface/metadata.cpp

```diff
@@ -79,12 +83,19 @@ class UserMetadataState {
 
 parthenon::internal::UserMetadataState metadata_state;
 
-MetadataFlag Metadata::AllocateNewFlag(std::string &&name) {
-  return metadata_state.AllocateNewFlag(std::move(name));
+MetadataFlag Metadata::AllocateNewFlag(const std::string &name) {
+  return metadata_state.AllocateNewFlag(name);
 }
 
 std::string const &MetadataFlag::Name() const { return metadata_state.FlagName(*this); }
 
+bool Metadata::FlagNameExists(const std::string &flagname) {
+  return (metadata_state.NamesToFlags().count(flagname) > 0);
+}
+MetadataFlag Metadata::FlagFromName(const std::string &flagname) {
```

Comment:
```suggestion
MetadataFlag Metadata::GetUserFlag(const std::string &flagname) {
```
This emphasizes both that it's the user flag (and the you should get Parthenon flags with `Metadata::Thing`) but also that it's not creating the flag, just retrieving it.

Reply:
I like this. It will be so.